src/gui/dialogs/preferences_dialog.cpp

```diff
@@ -248,22 +248,24 @@ void tpreferences::setup_combobox(
 		callback, options.second));
 }
 
+template <typename T, typename G>
 void tpreferences::setup_radio_toggle(
 		const std::string& toggle_id,
-		LOBBY_JOINS enum_value,
-		int start_value,
-		std::vector<std::pair<ttoggle_button*, int> >& vec,
+		const T& enum_value,
+		const int start_value,
+		G& group,
```

Comment:
I think you actually don't need the second template argument here - instead, you can use `tgroup<T>`.